Dockerfile

```diff
@@ -1,10 +1,29 @@
-FROM rust:slim as builder
-WORKDIR /usr/src/highlights
-COPY ./Cargo.toml ./Cargo.lock ./
-COPY ./src/ ./src
-RUN cargo install --path .
+FROM rust:1.49.0-slim-buster AS builder
+RUN USER=root cargo new --bin highlights && \
+    apt-get update && \
+    apt-get install -y --no-install-recommends musl-tools=1.1.21-2 pkg-config=0.29-6 libssl-dev=1.1.1d-0+deb10u4 && \
+    rustup target add x86_64-unknown-linux-musl && \
+    rustup component add rustfmt clippy && \
+    cargo install cargo-audit && \
+    mkdir highlights/.cargo && \
+    printf "[build]\ntarget = \"x86_64-unknown-linux-musl\"" > highlights/.cargo/config
```

Comment:
Could this be made compatible with ARM?

Reply:
Well sorta but not really. What I pushed _almost_ works but fails when the native target is a musl-ish. Going to try a fix but outside of that can just go to a glibc distro

Reply:
Thinking about it some more, I think it might be fine if this doesn't work on ARM. I suspect I'm going to be the primary user for the foreseeable future, and I'm not using docker to deploy anyway. If someone requests ARM docker support we can revisit.

Reply:
I feel like to get it working on ARM would require a better build system than cargo. It's pretty simple to even make a bash script that just looks at the Docker image architecture and picks the correct musl target